experimentation/export_model_predictions.py

```diff
@@ -0,0 +1,154 @@
+from collections.abc import Iterable
+from dataclasses import dataclass
+from itertools import islice
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from catboost import CatBoostClassifier, Pool
+from nxontology import NXOntology
+from sklearn.pipeline import Pipeline, make_pipeline
+
+from experimentation.model_utils import BiasedMaeMetric
+from nxontology_ml.data import get_efo_otar_slim, read_training_data
+from nxontology_ml.efo import NodeXrefFeatures
+from nxontology_ml.features import NodeInfoFeatures, PrepareNodeFeatures
+from nxontology_ml.model import CatBoostDataFormatter
+from nxontology_ml.text_embeddings.text_embeddings_transformer import (
+    TextEmbeddingsTransformer,
+)
+from nxontology_ml.utils import ROOT_DIR
+
+NON_DISEASE_THERAPEUTIC_AREAS: set[str] = {
```

Comment:
A more elegant way to handle this could be to:

1. Include disease TA terms as features
2. If a term does not have at least one disease TA term, then it cannot be classified
3. Remove any kind of filtering here

The spirit of this change would be to try to classify as much as possible unless there was something about a specific term that made it impossible, given the details of the features + models, rather than needing this somewhat awkward filter.

This isn't a strong preference.  I'll create an issue about adding TA features though.

---

FWIW,  I think this would be safer as an allowlist rather than a blocklist.  For a recent EFO OTAR version (~v3.53.0), the disease TA terms are:

```
term_label,term_id
psychiatric_disorder, MONDO:0002025
gastrointestinal_disease, EFO:0010282
immune_system_disease, EFO:0000540
endocrine_system_disease, EFO:0001379
infectious_disease, EFO:0005741
injury_poisoning_or_other_complication, OTAR:0000009
integumentary_system_disease, EFO:0010285
cardiovascular_disease, EFO:0000319
nutritional_or_metabolic_disease, OTAR:0000020
musculoskeletal_or_connective_tissue_disease, OTAR:0000006
urinary_system_disease, EFO:0009690
hematologic_disease, EFO:0005803
disorder_of_visual_system, MONDO:0024458
cancer_or_benign_tumor, MONDO:0045024
pregnancy_or_perinatal_disease, OTAR:0000014
nervous_system_disease, EFO:0000618
genetic_familial_or_congenital_disease, OTAR:0000018
reproductive_system_or_breast_disease, OTAR:0000017
disorder_of_ear, MONDO:0021205
respiratory_or_thoracic_disease, OTAR:0000010
```

Reply:
>I'll create an issue about adding TA features though.

https://github.com/related-sciences/nxontology-ml/issues/33

Reply:
Sounds good to me! 

> If a term does not have at least one disease TA term, then it cannot be classified

The way the pipeline is hooked up at the moment, once a record is in the feature pipeline, it will go through the model at the end of the pipeline. This can be changed by adding a filtering step somewhere in the pipeline (and need to update some assumptions where the number of records is constant).

I wonder if it would make sense to push this feature upstream to nx-ontology (similarly to the other topological properties). No strong opinion here.